src/hyperledger/smart-contracts/SmartContract.ts

```diff
@@ -0,0 +1,29 @@
+import { GatewayConnection } from "../connections/GatewayConnection";
+import { Contract, Gateway, GatewayOptions, Network, X509Identity } from 'fabric-network';
+import { IAuthenticatedMessageData, AuthenticatedMessage } from "../../torrente/messages/models/AuthenticatedMessage"
+import { PAYMENT_CHANNEL, PAYMENT_INTENTION_CONTRACT, REDEEM_CONTRACT } from '../config';
+
+export class SmartContract {
+
+    gatewayConnection : GatewayConnection;
+    contract : Contract;
+    smartContractName : string;
+    
+    public constructor (credentials: IAuthenticatedMessageData, smartContractName: string){
+        this.gatewayConnection = new GatewayConnection(credentials);
+        this.smartContractName = smartContractName;
+    }
+
+    public init = async () => {
+        this.contract = await this.gatewayConnection.getMicropaymentChaincodeReference(this.smartContractName);
+    };
+
+    protected invokeTransaction = async (name: string, args: any[]): Promise<any> => {
+        try {
+            return this.contract.submitTransaction(name, ...args);
```

Comment:
**talvez** a chamada correta seja this.contract.submitTransaction.apply(). Testar. Comparar com https://github.com/amazon-biobank/biobank/blob/72165070647e630d3531cdbc4f7d47b6b5a00e2f/application/contract/smartContract.js#L20